cumulus_etl/upload_notes/selector.py

```diff
@@ -48,28 +48,42 @@ def _create_docref_filter(
 
 def _filter_real_docrefs(docrefs_csv: str, docrefs: Iterable[dict]) -> Iterator[dict]:
     """Keeps any docrefs that match the csv list"""
-    with common.read_csv(docrefs_csv) as reader:
-        real_docref_ids = {row["docref_id"] for row in reader}
-
-    for docref in docrefs:
-        if docref["id"] in real_docref_ids:
-            yield docref
-
-            real_docref_ids.remove(docref["id"])
-            if not real_docref_ids:
-                break
-
+    transient = common.get_transient_progress()
+    found = 0
+    scanned = 0
+    with transient as progress:
+        task = progress.add_task(description = 'Getting docref ids', total=None)
+        with common.read_csv(docrefs_csv) as reader:
+            real_docref_ids = {row["docref_id"] for row in reader}
+
+        for docref in docrefs:
+            if docref["id"] in real_docref_ids:
+                found += 1
+                yield docref
+
+                real_docref_ids.remove(docref["id"])
+                if not real_docref_ids:
+                    break
+            scanned += 1
+            progress.update(task, description = f"Unmatched IDs: {len(real_docref_ids)}, found: {found}, scanned: {scanned}")
```

Comment:
minor nit: you'll miss the last progress update here if we early exit because we found all the target docrefs.